src/store/annotations/reducer.ts

```diff
@@ -38,7 +38,25 @@ const annotationsById = createReducer<AnnotationsState['byId']>({}, builder =>
         })
         .addCase(fetchAnnotationsAction.fulfilled, (state, { payload }) => {
             payload.entries.forEach(annotation => {
-                state[annotation.id] = addClientIds(annotation);
+                let newAnnotation = annotation;
+
+                if (annotation.target.type === 'drawing') {
+                    const {
```

Comment:
Can we avoid having any of this logic live in the main reducer? It seems like we'd just pass the entire annotation to a separate drawing-specific function, instead.